client/a8c-for-agencies/components/pending-payment-notification/index.tsx

```diff
@@ -1,54 +1,89 @@
-import { setPreference } from 'calypso/state/preferences/actions';
-import { getPreference } from 'calypso/state/preferences/selectors';
+import useFetchInvoices from 'calypso/a8c-for-agencies/data/purchases/use-fetch-invoices';
 import { CONTACT_URL_HASH_FRAGMENT } from '../a4a-contact-support-widget';
 import { A4A_INVOICES_LINK } from '../sidebar-menu/lib/constants';
 
 import './style.scss';
 
-const PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE = 'pending-payment-notification-dismissed';
+// TODO: Uncomment this if we would want to dismiss the notification.
+// const PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE = 'pending-payment-notification-dismissed';
 
 export default function PendingPaymentNotification() {
-	const dispatch = useDispatch();
-	const agency = useSelector( getActiveAgency );
-
-	const isDismissed = useSelector( ( state ) =>
-		getPreference( state, PENDING_PAYMENT_NOTIFICATION_DISMISS_PREFERENCE )
-	);
-
-	const canIssueLicenses = agency?.can_issue_licenses ?? true;
+	const invoices = useFetchInvoices( { starting_after: '', ending_before: '' }, undefined, 'paid' );
 
 	const translate = useTranslate();
 
-	// If the agency can issue licenses or the notification has been dismissed, do not show the notification.
-	if ( canIssueLicenses || isDismissed ) {
+	const daysDue = useMemo( () => {
+		const latestUnpaidInvoice =
+			invoices?.data?.items.reduce( ( latest, current ) => {
+				return latest < current.created ? latest : current.created;
+			}, Infinity ) || 0;
+		const nowInSeconds = Math.floor( Date.now() / 1000 ); // Get current time in seconds
+		const differenceInSeconds = nowInSeconds - latestUnpaidInvoice;
+		return Math.floor( differenceInSeconds / ( 60 * 60 * 24 ) ); // Convert seconds to days
+	}, [ invoices ] );
+	const daysLeft = 28 - daysDue;
+
+	if ( ! invoices?.isSuccess || ! invoices?.data?.items?.length || daysDue < 7 ) {
 		return null;
 	}
 
+	const title =
+		daysDue < 21 ? translate( 'Overdue invoice' ) : translate( 'Urgent: Overdue invoice' );
+
+	const level = daysDue < 21 ? 'warning' : 'error';
+
+	let description;
+
+	if ( daysDue < 14 ) {
+		description = translate(
+			'Your invoice is %(daysDue)d days overdue. Please pay as soon as possible to avoid service disruption.',
+			{
+				args: { daysDue },
+				comment: '%(daysDue)d is the number of days invoice is overdue.',
+			}
+		);
+	} else if ( daysDue < 21 ) {
+		description = translate(
+			'Your invoice is %(daysDue)d days overdue. You can’t issue new licenses until it’s paid.',
+			{
+				args: { daysDue },
+				comment: '%(daysDue)d is the number of days invoice is overdue.',
+			}
+		);
+	} else if ( daysDue < 27 ) {
+		description = translate(
+			'Your invoice is %(daysDue)d days overdue. We’ll revoke your active licenses if we don’t receive payment in %(daysLeft)d days.',
+			{
+				args: { daysDue, daysLeft },
+				comment: '%(daysDue)d is the number of days invoice is overdue.',
+			}
+		);
+	} else if ( daysDue < 29 ) {
```

Comment:
I wonder if this should be `daysDue < 28` instead so that on the 28th day we show the default "_Your product licenses have now been revoked..._" message rather than continuing to say they'll be revoked "_tomorrow_".